dbt/task/compile.py

```diff
@@ -81,29 +58,36 @@ def __wrap_in_create(self, path, query, model_config):
 
         return create_template.format(**opts)
 
+    def __get_model_identifiers(self, model_filepath):
+        model_group = os.path.dirname(model_filepath)
+        model_name, _ = os.path.splitext(os.path.basename(model_filepath))
+        return model_group, model_name
 
-    def __get_sql_file_config(self, src_path, f):
-        model_path = os.path.join(src_path, os.path.dirname(f))
-        config = self.model_configs.get(model_path, default_model_config)
-        identifier, ext = os.path.splitext(os.path.basename(f))
-        model_config = config.copy()
+    def __get_model_config(self, model_group, model_name):
+        """merges model, model group, and base configs together. Model config
+        takes precedence, then model_group, then base config"""
 
-        if identifier in model_config:
-            model_config.update(config[identifier])
+        model_configs = self.project['models']
 
-        return model_config
+        config = model_configs['base'].copy()
```

Comment:
I don't like that "base" is a magic word here, it means a model/model group can't be called that.  Some alternative proposals:

 - Don't allow this to be set hierarchically.   Either only allow a global setting, or only allow per-model settings.
 - Use regexps to specify the models to apply the settings to.  In that case, `"base"` would become `"*"`.